config/parser.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+type parser struct {
+	dups map[Source]string
+}
+
+func newParser() *parser {
+	return &parser{}
+}
+
+func (p *parser) GetFields(cfg interface{}) ([]*Field, error) {
```

Comment:
I think a more proper, conceptual, name for that function would be `ParseCfg`

Reply:
Hi, in general I think it is a good idea to name functions in a specific, self-explanatory way. So the next developer will be able to understand the purpose of this function without a deep dive. 

Right now this method does only one thing - it gets fields from configuration structure. In future this parser can be extended to do a lot more things, but on other side it with high probability will change a return type as well. And at that point of time function name, return type and may be parameters list will be changed to reflect a new function purpose. Or (what probably better idea) - a new more general function will be introduced.

In the end it highly depends on naming convention that you use in the team. Of course this function can be renamed to be more general, but I think we need to consider all the drawbacks that I've wrote above. What do you think?

Reply:
I agree on your point for self-explanatory naming and that's why I a mentioned to change it. The parser **parses** configuration struct and returns fields. The response type doesn't define the process itself.
You can also see that in [Go's parser](https://golang.org/pkg/go/parser/) package. The return type doesn't alter the point that the function parses something to gather it (ex. `ParseDir`, `ParseExpr` etc.)

Reply:
Ah, I got your point. Changed in new commit. Thank you for review!